src/folsom_graphite_sender.erl

```diff
@@ -43,18 +46,19 @@ send(Message) ->
 %% gen_server Function Definitions
 %% ------------------------------------------------------------------
 
-init([GraphiteHost, GraphitePort]) ->
-    case gen_tcp:connect(GraphiteHost, GraphitePort,
-                         [binary, {packet, raw}, {active, false},
-                          {keepalive, true}, {nodelay, true},
-                          {send_timeout, 2000}], ?CONNECT_TIMEOUT) of
-        {ok, Socket} ->
-            State = #state{socket = Socket},
-            {ok, State};
-        {error, Reason} ->
-            lager:error("Failed to connect to graphite server on ~s:~B: ~p~n", [GraphiteHost, GraphitePort, Reason]),
-            {stop, {error, failed_connect}}
+init([GraphiteHost, GraphitePort, RetryInterval]) ->
+    State = #state{host = GraphiteHost, port = GraphitePort, retry_interval = RetryInterval},
+    case connect(State) of
+        #state{socket = undefined, retry_interval = 0} ->
+            % Only reconnect when retry_interval is set
+            {stop, {error, failed_connect}};
+        #state{retry_interval = Interval} = State2 ->
+            timer:send_after(Interval, reconnect),
```

Comment:
Interestingly, send_after/2 doesn't seem to exist even though it's referenced in that doc  - only send_after/3 is documented.